### PR TITLE
[TensorExpr] Add lowering for aten::max (reduction).

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66847
* #66846
* #66527
* #66522
* #66515
* #66514
* #66523
* __->__ #66519
* #66518
* #66517
* #66516

Differential Revision: [D31590853](https://our.internmc.facebook.com/intern/diff/D31590853)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31590853).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31590853).

**Review by eellison — COMMENTED** [review]
How close are we to codegen'ing this ? I know @navahgar  was at one point looking at reductions

**Comment by ZolotukhinM** [comment]
> How close are we to codegen'ing this ? I know @navahgar was at one point looking at reductions

I think we can codegen it now, it's just safer to go with an external call in the beginning since it would work for sure (whereas if I try to go with a native lowering I might screw up perf and/or correctness).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31590853).

File: torch/csrc/jit/tensorexpr/external_functions.cpp [hunk 1]
```diff
@@ -384,6 +384,28 @@ void nnc_aten_mean(
...
+    int8_t* buf_dtypes,
+    int64_t args_num,
+    int64_t* extra_args) {
+  std::vector<at::Tensor> tensors =
+      constructTensors(bufs_num, buf_data, buf_ranks, buf_dims, buf_dtypes);
+
+  at::Tensor& r = tensors[0];
+  const at::Tensor& x = tensors[1];
+  int64_t max_dim = extra_args[0];
+  bool keep_dim = extra_args[1];
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/external_functions.cpp:401]: Do we need a cast to `bool` here? Just to be explicit.

**Review by navahgar — APPROVED** [review]
LGTM

**Pull request closed without merging**